packages/node-core/src/indexer/fetch.service.ts

```diff
@@ -266,10 +250,17 @@ export abstract class BaseFetchService<
         : initBlockHeight;
     };
 
-    if (this.useFatDictionary && this.useDictionary && this.dictionaryService.startHeight > getStartBlockHeight()) {
+    // If currently there is a dictionary but metadata is not valid
+    // to find next dictionary
+    if (this.dictionaryService.useDictionary && !this.dictionaryService.dictionary.metadataValid) {
```

Comment:
I think `useDictionary` should be adding the check for `!this.dictionaryService.dictionary.metadataValid` internally

Reply:
`useDictionary` already check `dictionaryService.dictionary.metadataValid`, I think this should be `!this.dictionaryService.useDictionary`.